src/main/java/uk/gov/hmcts/reform/judicialbooking/domain/service/common/PrepareDataService.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.judicialbooking.domain.service.common;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.judicialbooking.controller.advice.exception.ResourceNotFoundException;
+import uk.gov.hmcts.reform.judicialbooking.data.BookingEntity;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.Appointment;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.Authorisation;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.BookingResponse;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.JudicialUserProfile;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBooking;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBookingAssignmentsRequest;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBookingRequest;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.enums.Status;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class PrepareDataService {
+
+    public BookingEntity prepareBooking(BookingEntity booking, List<JudicialUserProfile> judicialUserProfiles) {
+
+        List<Appointment> appointment = judicialUserProfiles.get(0).getAppointments().stream()
+                .filter(appointment1 -> appointment1.getAppointmentId().equals(booking.getAppointmentId()))
+                .collect(Collectors.toList());
+
+        if (appointment.isEmpty()) {
+            throw new ResourceNotFoundException(
+                    "Appointment retrieved from JRD is empty, your appointmentId may be invalid");
+        }
+
+        return prepareBookingVars(appointment, booking);
+    }
+
+    public BookingEntity prepareBookingVars(List<Appointment> appointment, BookingEntity booking) {
+        if (StringUtils.isBlank(booking.getRoleId())) {
+            booking.setRoleId(appointment.get(0).getRoleId());
+        }
+        if (StringUtils.isBlank(booking.getBaseLocationId())) {
+            booking.setBaseLocationId(appointment.get(0).getBaseLocationId());
+        }
+        booking.setContractTypeId(appointment.get(0).getContractTypeId());
+        //TODO booking.setRegionId(); this one requires another endpoint call I think
+        booking.setStatus(Status.NEW.toString());
+
+        return booking;
+    }
+
+    public OrmBookingAssignmentsRequest prepareOrmBookingRequest(BookingEntity booking,
+                                                                 List<JudicialUserProfile> judicialUserProfiles) {
```

Comment:
Same as above, we don't need to deal with list of judicialUserProfile, pls update that to single object.